frontend/plan/src/components/selector/CourseList.js

```diff
@@ -31,22 +31,41 @@ const courseSort = (courses, sortMode) => {
     return sorted;
 };
 
-export default function CourseList({
-    courses, getCourse, sortMode,
-}) {
+const CourseList = ({
+    courses, getCourse, sortMode, scrollPos, setScrollPos,
+}) => {
+    const listRef = useRef(null);
+    useEffect(() => {
+        // If we have a snapshot value, we've just added new items.
+        // Adjust scroll so these new items don't push the old ones out of view.
+        // (snapshot here is the value returned from getSnapshotBeforeUpdate)
+        listRef.current.scrollTop = scrollPos;
+        // Return cleanup function
+        return () => setScrollPos(listRef.current.scrollTop);
```

Comment:
This seems to be causing an infinite rerender loop since useEffect has no deps and the cleanup function gets called every rerender

Reply:
updated deps